File: presentation/src/main/java/com/study/platform/domain/notification/api/SseEmitterAdapter.java
package com.study.platform.domain.notification.api;

import com.study.platform.domain.notification.model.SseConnection;
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

import java.io.IOException;

public class SseEmitterAdapter implements SseConnection {

    private final SseEmitter emitter;

    public SseEmitterAdapter(SseEmitter emitter) {
        this.emitter = emitter;
    }

    @Override
    public void send(String eventName, Object data) throws IOException {
        emitter.send(SseEmitter.event().name(eventName).data(data));
    }
}

File: presentation/src/main/java/com/study/platform/domain/notification/api/SseEmitterAdapter.kt
package com.study.platform.domain.notification.api

import com.study.platform.domain.notification.model.SseConnection
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter
import java.io.IOException

class SseEmitterAdapter(
    private val emitter: SseEmitter
) : SseConnection {

    @Throws(IOException::class)
    override fun send(eventName: String, data: Any) {
        emitter.send(SseEmitter.event().name(eventName).data(data))
    }
}
